### PR TITLE
Support for optimum-intel models

### PR DESCRIPTION
**Pull request by slyalin**
Support for models from optimum-intel: they are loaded and converted on-the-fly without relying on custom modeling which exists in vLLM. To enable optimum-intel conversion set environment variable VLLM_OPENVINO_OPTIMUM to 1 (while not setting or setting to 0 will switch to vLLM modeling and disable optimum-intel usage).

To make an optimum-intel model compatible with vLLM kv-cache processing and accept vLLM specific parameters, the model is being converted to OpenVINO format inside optimum-intel first and then a number of transformations are applied to produce vLLM compatible form:

1. ReadValue -> ... -> SDPA pattern with several variations is recognized and replaced by PagedAttentionExtension with a few gluing nodes.
2. attention_mask input if exists in the original model and used for position_ids generation is replaced by a dedicated position_ids parameter.
3. Specific patterns that get max_context_len is replaced by a dedicated parameter.
4. Automatically enabled weights compression that usually happens while exporting big models from optimum-intel, is temporary disabled due to visible difference in output which complicates accuracy validation.

vLLM Sampler class is changed because it now accepts logits instead of hidden_states (due to a difference in how the ending MatMul is implemented -- another optimization opportunity that is worth fusing into OV model).

This is a limited support because too narrow patterns are used in transformations and there are explicit and intentional limitations in the code for the purpose of simplification (see TODOs). The glue nodes that make tensor layouts compatible around PageAttentionExtension are not optimized as a part of this PR with the hope (not confirmed) that they will be optimized in compile_model if possible (e.g. repeated Transposes).

Validated manually with:
- opt-125, tiny-llama, mistral-7b, EleutherAI/gpt-j-6b (matches vLLM original modeling also in OV),
- Qwen/Qwen1.5-0.5B-Chat (generate consistently, no comparison as no vLLM custom modeling for Qwen2 arch; based on not yet committed https://github.com/huggingface/optimum-intel/pull/568).

### CODE REVIEW

File: vllm/worker/model_runner.py [hunk 1]
```diff
@@ -236,58 +281,230 @@ def wrapper(module, target_op, *args, **kwargs):
...
+            def callback(m: Matcher) -> bool:
+                mapping = m.get_pattern_value_map()
+                if not position_ids_parameter:
+                    position_ids_parameter.append(opset13.parameter(shape=[-1, -1], dtype=np.int64, name="position_ids"))
+                    print('CREATED A NEW position_ids PARAMETER')
+                replace_node(mapping[position_ids].get_node(), position_ids_parameter[0])
+                print('APPLIED position_ids PARAMETER INSTEAD OF attention_mask-BASED SUB-GRAPH')
+                return True
+
+            self.register_matcher(Matcher(add, "InputAndPoistionIDsAdd"), callback)
```
Comment (ilya-lavrenov) [inline vllm/worker/model_runner.py:474]: InputAndPositionIDsAdd
